### PR TITLE
Build wasm with some extra flags so c++ EH works. Remove obsolete wasm options.

### PR DESCRIPTION
**Pull request by vargaz**

### CODE REVIEW

File: src/mono/wasm/Makefile [hunk 1]
```diff
@@ -62,7 +62,7 @@ MONO_LIBS = \
 	$(ICU_LIBDIR)/libicuuc.a \
 	$(ICU_LIBDIR)/libicui18n.a
 
-EMCC_FLAGS=--profiling-funcs -s WASM=1 -s ALLOW_MEMORY_GROWTH=1 -s BINARYEN=1 -s ALIASING_FUNCTION_POINTERS=0 -s NO_EXIT_RUNTIME=1 -s "EXTRA_EXPORTED_RUNTIME_METHODS=['ccall', 'FS_createPath', 'FS_createDataFile', 'cwrap', 'setValue', 'getValue', 'UTF8ToString', 'UTF8ArrayToString', 'addFunction']" -s "EXPORTED_FUNCTIONS=['_putchar']" --source-map-base http://example.com  -emit-llvm -s FORCE_FILESYSTEM=1 -s USE_ZLIB=1
+EMCC_FLAGS=--profiling-funcs -s DISABLE_EXCEPTION_CATCHING=0 -s ALLOW_MEMORY_GROWTH=1 -s NO_EXIT_RUNTIME=1 -s USE_ZLIB=1  -s FORCE_FILESYSTEM=1 -s "EXTRA_EXPORTED_RUNTIME_METHODS=['ccall', 'FS_createPath', 'FS_createDataFile', 'cwrap', 'setValue', 'getValue', 'UTF8ToString', 'UTF8ArrayToString', 'addFunction']" -s "EXPORTED_FUNCTIONS=['_putchar']" --source-map-base http://example.com  -emit-llvm
```
Comment (akoeplinger) [inline src/mono/wasm/Makefile:65]: are we sure we want to turn this on by default, it apparently has perf implications: https://github.com/emscripten-core/emscripten/blob/6a71998ad495d911876a9d17e25409c9899eacfe/src/settings.js#L558-L559

Comment (jeromelaban) [reply src/mono/wasm/Makefile:65]: It was enabled by default in mono/mono, and the performance impact was not noticeable at the time. Currently, it's a pain to debug standard scenarios without it.

Comment (vargaz) [reply src/mono/wasm/Makefile:65]: Without it, exception handling in AOT will not work.

Comment (vargaz) [reply src/mono/wasm/Makefile:65]: We can add it to the aot build instead if needed.

**Review by akoeplinger — APPROVED** [review]

**Comment by vargaz** [comment]
Added back the option to WasmSample.csproj.

**Pull request merged**